wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,342 @@
+			self::OPTION_GROUP,
+			self::UPGRADE_HTTPS_OPTION,
+			array(
+				'type'              => 'string',
+				'sanitize_callback' => array( $this, 'upgrade_https_sanitize_callback' ),
+			)
+		);
+	}
+
+	/**
+	 * Sanitization callback for the upgrade HTTPS option.
+	 *
+	 * @param string $raw_value The value to sanitize.
+	 * @return bool Whether the option is true or false.
+	 */
+	public function upgrade_https_sanitize_callback( $raw_value ) {
+		unset( $raw_value );
+		return isset( $_POST[ self::UPGRADE_HTTPS_OPTION ] ); // WPCS: CSRF OK.
+	}
+
+	/**
+	 * Adds a settings field to the 'General Settings' page.
+	 *
+	 * Only add this if the site can support HTTPS,
+	 * but the home and siteurl option values are not HTTPS (WordPress Address and Site Address).
+	 * This UI would not apply if those URLs are already HTTPS.
+	 */
+	public function add_settings_field() {
+		/*
+		 * Todo: add ! before $this->is_currently_https(), as this is only for development.
+		 * It allows developing this while the siteurl and home values are HTTPS.
+		 */
+		if ( ! $this->wp_https_detection->is_currently_https() ) {
+			add_settings_field(
+				self::HTTPS_SETTING_ID,
+				__( 'HTTPS', 'pwa' ),
+				array( $this, 'render_https_settings' ),
+				self::OPTION_GROUP
+			);
+		}
+	}
+
+	/**
+	 * Renders the HTTPS settings in /wp-admin on the General Settings page.
+	 */
+	public function render_https_settings() {
+		$upgrade_https_value = (bool) get_option( self::UPGRADE_HTTPS_OPTION );
+		$https_more_details  = sprintf(
+			'<a href="%s">%s</a>',
+			__( 'https://make.wordpress.org/support/user-manual/web-publishing/https-for-wordpress/', 'pwa' ),
+			__( 'More details', 'pwa' )
+		);
+
+		?>
+		<p class="description">
+			<?php
+			echo wp_kses_post( sprintf(
+				/* translators: %s: a link for more details */
+				__( 'HTTPS is essential to securing your WordPress site, we strongly suggest upgrading to it. %s', 'pwa' ),
+				$https_more_details
+			) );
+			?>
+		</p>
+		<p>
+			<label><input name="<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>" type="checkbox" <?php checked( $upgrade_https_value ); ?> value="<?php echo esc_attr( self::OPTION_CHECKED_VALUE ); ?>"><?php esc_html_e( 'Upgrade to secure connection', 'pwa' ); ?></label>
+		</p>
+		<script>
+			(function ( $ ) {
+				// Move this UI under the Site Address (URL) <tr> on the General Settings page.
+				$( 'input[name=<?php echo esc_attr( self::UPGRADE_HTTPS_OPTION ); ?>]' ).parents( 'tr' ).insertAfter( $( 'label[for=home]' ).parents( 'tr') );
+			})( jQuery );
+		</script>
+		<?php
+
+		$insecure_urls_option = get_option( WP_HTTPS_Detection::INSECURE_CONTENT_OPTION_NAME );
+		$all_insecure_urls    = $insecure_urls_option;
+		$total_urls_count     = count( $all_insecure_urls );
+		$insecure_content_id  = 'insecure-content';
+		$show_more_button_id  = 'view-urls';
+		$insecure_urls_class  = 'insecure-urls';
+
+		// Exit if there is no insecure URL to display.
+		if ( ! $total_urls_count ) {
```

Comment:
I'd suggest moving this guard clause up to line 177.  If there are no insecure URLs, bail out; else, then set the other `insecure_content_id` and other variable values and continue.

Reply:
Sure, https://github.com/xwp/pwa-wp/pull/36/commits/fc159a274426fae4d906fe596b39df86d0f4eb8a this `return` statement up.